### PR TITLE
Bugfix/dtserwthre 232 wrap truncated headers

### PR DESCRIPTION
**Pull request by nanbehera**
wrap truncated headers for large title

### CODE REVIEW

**Review by pdamodarannair — APPROVED** [review]

**Review by aparthibanpaypal — APPROVED** [review]

File: Common/Sources/Extensions/UIViewController.swift [hunk 1]
```diff
@@ -31,6 +31,14 @@ public extension UIViewController {
...
             self.navigationController?.navigationBar.prefersLargeTitles = true
             currentNavigationItem.largeTitleDisplayMode = mode
+            if #available(iOS 13.0, *) {
+                ThemeManager.applyToUINavigationBar()
+            } else {
+              self.navigationController?.navigationBar.largeTitleTextAttributes = [
+                NSAttributedString.Key.foregroundColor: Theme.NavigationBar.largeTitleColor,
+                NSAttributedString.Key.font: UIFont.boldSystemFont(ofSize: 18.0)
+              ]
+            }
```
Comment (jsaini-hw) [inline Common/Sources/Extensions/UIViewController.swift:41]: @nanbehera there's a reason why `ThemeManager.applyToUINavigationBar()` is optional because not every integrator wants our custom navigation bar implementation. So you have to remove this logic from here

File: Common/Sources/Helper/ThemeManager.swift [hunk 2]
```diff
@@ -59,7 +59,7 @@ public class ThemeManager: NSObject {
             navBarAppearance.largeTitleTextAttributes =
                 [
                     NSAttributedString.Key.foregroundColor: Theme.NavigationBar.largeTitleColor,
-                    NSAttributedString.Key.font: Theme.NavigationBar.largeTitleFont
+                    NSAttributedString.Key.font: UIFont.boldSystemFont(ofSize: 20.0)
```
Comment (jsaini-hw) [inline Common/Sources/Helper/ThemeManager.swift:62]: I thought the idea was to use sizeToFit() instead of using smaller font size

File: Common/Sources/Extensions/UIViewController.swift (hunk 1 above)
Comment (nanbehera) [reply Common/Sources/Extensions/UIViewController.swift]: Ok . Code reverted back

File: Common/Sources/Helper/ThemeManager.swift (hunk 2 above)
Comment (nanbehera) [reply Common/Sources/Helper/ThemeManager.swift:62]: Actually sizeToFit() can use navigationbar titleView, which will come in center. So i have choosed this smaller font size.

File: Common/Sources/Helper/ThemeManager.swift [hunk 3]
```diff
@@ -46,7 +46,7 @@ public class ThemeManager: NSObject {
             proxy.largeTitleTextAttributes =
                 [
                     NSAttributedString.Key.foregroundColor: Theme.NavigationBar.largeTitleColor,
-                    NSAttributedString.Key.font: Theme.NavigationBar.largeTitleFont
+                    NSAttributedString.Key.font: UIFont.boldSystemFont(ofSize: 17.0)
```
Comment (jsaini-hw) [inline Common/Sources/Helper/ThemeManager.swift:49]: If you are going to hardcode this value here, then what will happen if Integrators wants to change the font size and set `Theme.NavigationBar.largeTitleFont = UIFont.boldSystemFont(ofSize: 21.0)` but since we are not using Theme.NavigationBar.largeTitleFont here so how do you think that will work?

File: Common/Sources/Helper/Theme.swift [hunk 4]
```diff
@@ -65,9 +65,9 @@ public class Theme: NSObject {
         public static var backButtonColor = Theme.tintColor
         @available(iOS 11.0, *)
         /// The `UINavigationBar` large title font
-        public static var largeTitleFont = UIFont.preferredFont(forTextStyle: .largeTitle)
+        public static var largeTitleFont = UIFont.boldSystemFont(ofSize: 20.0)
         /// The `UINavigationBar` title font
-        public static var titleFont = UIFont.preferredFont(forTextStyle: .body)
+        public static var titleFont = UIFont.boldSystemFont(ofSize: 17.0)
```
Comment (jsaini-hw) [inline Common/Sources/Helper/Theme.swift:70]: .body is already `
Body | Regular | 17
-- | -- | --
`
Why did you change it to 17?

File: Common/Sources/Helper/ThemeManager.swift [hunk 5]
```diff
@@ -46,7 +46,7 @@ public class ThemeManager: NSObject {
             proxy.largeTitleTextAttributes =
                 [
                     NSAttributedString.Key.foregroundColor: Theme.NavigationBar.largeTitleColor,
-                    NSAttributedString.Key.font: Theme.NavigationBar.largeTitleFont
+                    NSAttributedString.Key.font: Theme.NavigationBar.titleFont
```
Comment (jsaini-hw) [inline Common/Sources/Helper/ThemeManager.swift:49]: what's happening if you use largeTitleFont instead of titleFont because you have decreased that to 20 as well

File: Common/Sources/Helper/Theme.swift (hunk 4 above)
Comment (nanbehera) [reply Common/Sources/Helper/Theme.swift:70]: okay.will update

File: Common/Sources/Helper/ThemeManager.swift (hunk 5 above)
Comment (nanbehera) [reply Common/Sources/Helper/ThemeManager.swift:49]: Code updated for SE 1st Gen

**Review by jsaini-hw — APPROVED** [review]

**Review by malaw-hw — APPROVED** [review]

**Comment by jsaini-hw** [comment]
Travis ci failing due to flaky tests so merging the PR using admin rights.
